libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -358,7 +367,8 @@ MAV_RESULT AP_Mount::handle_command_do_gimbal_manager_pitchyaw(const mavlink_com
         return MAV_RESULT_ACCEPTED;
     }
 
-    return MAV_RESULT_FAILED;
+    // If this was a yaw lock/unlock only command, we need to return MAV_RESULT_ACCEPTED
+    return yaw_lock_changed ? MAV_RESULT_ACCEPTED : MAV_RESULT_FAILED;
```

Comment:
I think we should return success as long as the actual yaw lock state is in the requested state.  So it shouldn't matter if it was changed or not.